velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -972,10 +984,13 @@ findFloatingPointColumns(const MaterializedRow& row) {
 // hence can be compared directly.
 bool assertEqualResults(
     const MaterializedRowMultiset& expectedRows,
+    const TypePtr& expectedType,
     const MaterializedRowMultiset& actualRows,
+    const TypePtr& actualType,
     const std::string& message) {
   if (expectedRows.empty() != actualRows.empty()) {
-    ADD_FAILURE() << generateUserFriendlyDiff(expectedRows, actualRows)
+    ADD_FAILURE() << generateUserFriendlyDiff(
```

Comment:
If expectedType is empty, since expectedRows were empty, it means there are actual rows. We need to pass actualRows here.
Can you add a test for this?

Reply:
We are already passing actualRows here, right?
```
    ADD_FAILURE() << generateUserFriendlyDiff(
                         expectedRows, actualRows, expectedType)
```

Reply:
Fixed. Passed actualType when expectedRows = 0 and actualRows > 0.